apps/submission-service/src/submission.controller.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Inject,
+  Post,
+  Query,
+  UseInterceptors,
+} from '@nestjs/common';
+import { SubmissionService } from './submission.service';
+import {
+  ApiBearerAuth,
+  ApiBody,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { KAFKA_CLIENT_NAME, KafkaService } from '@app/kafka';
+import { CreateSubmissionDto } from '@app/types';
+import { TargetsService } from './targets/targets.service';
+import { Payload } from '@nestjs/microservices';
+import { FileInterceptor } from '@nestjs/platform-express';
+
+@ApiTags('Submission Controller')
+@Controller()
+export class SubmissionController {
+  constructor(
+    private readonly submissionService: SubmissionService,
+    private readonly targetsService: TargetsService,
+    @Inject(KAFKA_CLIENT_NAME) private readonly kafkaService: KafkaService,
+  ) {}
+
+  @Get('/all')
+  @ApiBearerAuth('access-token')
+  @ApiOperation({ summary: 'Get all submissions' })
+  @ApiResponse({ status: 200 })
+  @ApiResponse({ status: 500, description: 'Internal server error' })
+  async getAll(
+    @Query('targetUuid') targetUuid: string,
+    @Query('userUuid') userUuid: string,
+  ) {
+    const targetResult = await this.targetsService.findOne(targetUuid);
+    if (!targetResult.data) {
+      return { message: `Target with UUID ${targetUuid} not found.` };
+    }
+
+    const submissionResult = await this.submissionService.findByTarget(
+      targetResult.data.uuid,
+    );
+
+    // If user is the owner of the target, return all submissions
+    if (userUuid === targetResult.data.ownerUuid) {
+      return submissionResult.data;
+    }
+
+    // Otherwise, return only their own submissions (if any)
+    const userSubmissions = submissionResult.data.filter(
+      (submission: { ownerUuid: string }) => submission.ownerUuid === userUuid,
+    );
+
+    if (userSubmissions.length > 0) {
+      return userSubmissions;
+    }
+
+    return {
+      message: `Access denied. You do not own the target or any of its submissions.`,
+    };
+  }
+
+  @Post()
+  @ApiOperation({
+    summary: 'Create a new submission',
+    description:
+      'Creates a new submission with provided data. All fields are required.',
+  })
+  @ApiBody({ type: CreateSubmissionDto })
+  @ApiResponse({ status: 201, description: 'Submission created successfully' })
+  @ApiResponse({ status: 400, description: 'Invalid request body' })
+  @ApiResponse({ status: 500, description: 'Internal server error' })
+  @UseInterceptors(FileInterceptor('image'))
+  async create(@Body() createSubmissionDto: CreateSubmissionDto) {
+    const targetResult = await this.targetsService.findOne(
+      createSubmissionDto.targetUuid,
+    );
+
+    if (targetResult.data.closedForSubmission) {
```

Comment:
```suggestion
    if (targetResult?.data?.closedForSubmission) {
```

^ wel nog testen of dit werkt. Als de target niet in de submission-database zit, kan dit een error geven. Ik had al targets toegevoegd (voor het fetchen) en kreeg errors omdat ze daar nog niet stonden

Reply:
En werkt het?

Reply:
Ik zal het testen haha